cluster/partitioner/partitioner.go

```diff
@@ -13,40 +12,43 @@ type Partitioner interface {
 
 type Kafka struct {
 	PartitionBy string
-	Partitioner sarama.Partitioner
 }
 
 func NewKafka(partitionBy string) (*Kafka, error) {
 	switch partitionBy {
 	case "byOrg":
 	case "bySeries":
+	case "bySeriesWithTags":
 	default:
-		return nil, fmt.Errorf("partitionBy must be one of 'byOrg|bySeries'. got %s", partitionBy)
+		return nil, fmt.Errorf("partitionBy must be one of 'byOrg|bySeries|bySeriesWithTags'. got %s", partitionBy)
 	}
 	return &Kafka{
 		PartitionBy: partitionBy,
-		Partitioner: sarama.NewHashPartitioner(""),
 	}, nil
 }
 
 func (k *Kafka) Partition(m schema.PartitionedMetric, numPartitions int32) (int32, error) {
-	key, err := k.GetPartitionKey(m, nil)
+	partition, err := k.GetPartition(m, numPartitions)
```

Comment:
can we not just simply `return k.GetPartition(m, numPartitions)`

Reply:
Which then leaves the question, why to we even need this function.  Cant users just call GetPartition() directly  (perhaps renaming it to just Partition())